SftpLibsshAdapter/SFTPConnection.cpp

```diff
@@ -0,0 +1,269 @@
+#include "stdafx.h"
+
+#include <stdlib.h>
+#include <errno.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <exception> 
+
+#include "SFTPConnection.h"
+
+namespace systelab { namespace sftp {
+
+	namespace
+	{
+		// Used by function "sftp_open", parameter mode_t "mode"
+		// The mode_t accepted values are defined when #include <sys/types.h> but only for Linux (e.g.: S_IRWXU)
+		const mode_t  DEFAULT_FILE_PERMISSION = 0; 
+
+		std::string readFileContent(const std::string filename)
+		{
+			std::string content = "";
+			try
+			{
+				std::ifstream file(filename);
+
+				file.seekg(0, std::ios::end);
+				content.reserve((unsigned int)file.tellg());
+				file.seekg(0, std::ios::beg);
+
+				content.assign((std::istreambuf_iterator<char>(file)), std::istreambuf_iterator<char>());
+			}
+			catch (...) {}
+			return content;
+		}
+
+	}
+
+	SFTPConnection::~SFTPConnection()
+	{
+		close();
+	}
+
+	void SFTPConnection::connect(const std::string& ip,
+									unsigned int port,
+									const std::string& username,
+									const std::string& pubKeyFile,
+									const std::string& privKeyFile,
+									const std::function<std::string()>& getPrivKeyPassPhraseFn,
+									const std::vector<std::string>& serverFingerPrints)
+	{
+		try
+		{
+			sshConnect(ip, port);
+			verifySFTPServerIdentity(serverFingerPrints);
+			sshAuthorize(username, pubKeyFile, privKeyFile, getPrivKeyPassPhraseFn());
+			sftpConnect();
+		}
+		catch (ISFTPClientAdapter::Exception exc)
```

Comment:
If you are doing exception handling, do it as const& to avoid copying the exception object